tests/components/group/test_init.py

```diff
@@ -18,307 +17,375 @@
+
+    test_group = await group.Group.async_create_group(
+        hass, "init_group", ["light.Bowl", "light.Ceiling"], False
+    )
+
+    assert sorted(["light.ceiling", "light.bowl"]) == sorted(
+        group.expand_entity_ids(hass, [test_group.entity_id])
+    )
+
+
+async def test_expand_entity_ids_does_not_return_duplicates(hass):
+    """Test that expand_entity_ids does not return duplicates."""
+    hass.states.async_set("light.Bowl", STATE_ON)
+    hass.states.async_set("light.Ceiling", STATE_OFF)
+
+    assert await async_setup_component(hass, "group", {})
+
+    test_group = await group.Group.async_create_group(
+        hass, "init_group", ["light.Bowl", "light.Ceiling"], False
+    )
+
+    assert ["light.bowl", "light.ceiling"] == sorted(
+        group.expand_entity_ids(hass, [test_group.entity_id, "light.Ceiling"])
+    )
+
+    assert ["light.bowl", "light.ceiling"] == sorted(
+        group.expand_entity_ids(hass, ["light.bowl", test_group.entity_id])
+    )
+
+
+async def test_expand_entity_ids_recursive(hass):
+    """Test expand_entity_ids method with a group that contains itself."""
+    hass.states.async_set("light.Bowl", STATE_ON)
+    hass.states.async_set("light.Ceiling", STATE_OFF)
+
+    assert await async_setup_component(hass, "group", {})
+
+    test_group = await group.Group.async_create_group(
+        hass,
+        "init_group",
+        ["light.Bowl", "light.Ceiling", "group.init_group"],
+        False,
+    )
+
+    assert sorted(["light.ceiling", "light.bowl"]) == sorted(
+        group.expand_entity_ids(hass, [test_group.entity_id])
+    )
+
+
+async def test_expand_entity_ids_ignores_non_strings(hass):
+    """Test that non string elements in lists are ignored."""
+    assert [] == group.expand_entity_ids(hass, [5, True])
+
+
+async def test_get_entity_ids(hass):
+    """Test get_entity_ids method."""
+    hass.states.async_set("light.Bowl", STATE_ON)
+    hass.states.async_set("light.Ceiling", STATE_OFF)
+
+    assert await async_setup_component(hass, "group", {})
+
+    test_group = await group.Group.async_create_group(
+        hass, "init_group", ["light.Bowl", "light.Ceiling"], False
+    )
+
+    assert ["light.bowl", "light.ceiling"] == sorted(
+        group.get_entity_ids(hass, test_group.entity_id)
+    )
+
+
+async def test_get_entity_ids_with_domain_filter(hass):
+    """Test if get_entity_ids works with a domain_filter."""
+    hass.states.async_set("switch.AC", STATE_OFF)
+
+    assert await async_setup_component(hass, "group", {})
+
+    mixed_group = await group.Group.async_create_group(
+        hass, "mixed_group", ["light.Bowl", "switch.AC"], False
+    )
+
+    assert ["switch.ac"] == group.get_entity_ids(
+        hass, mixed_group.entity_id, domain_filter="switch"
+    )
+
+
+async def test_get_entity_ids_with_non_existing_group_name(hass):
+    """Test get_entity_ids with a non existing group."""
+    assert [] == group.get_entity_ids(hass, "non_existing")
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6372789)